DMTN-049.tex

```diff
@@ -122,40 +124,69 @@ \subsection{Federating a User-Generated Photo-$z$ Catalog}\label{ssec:time_ops_u
+The DM team endeavors to be clear about the nature of these technical issues, but the following is subject to change as the DM System evolves and as feedback from the community is considered. 
+Adherence with these technical considerations is one of the {\it proposed} evaluation criteria in \S~\ref{sec:eval}.
+
+\textbf{Scalability:}
+The strongest point of consideration for any contributed  photo-$z$ estimator will be its ability to run at LSST scale. 
+The LSST Object catalog is projected to contain $\approx$ 37 billion {\tt Objects} by the end of the survey;  photo-$z$ estimators must be able to efficiently compute a photometric redshift for all {\tt Objects} in the DR Object catalog \footnote{An estimator will be considered efficient if it runs in a reasonable amount of time on the available compute resources. }. 
+Shortlisted photo-$z$ estimators will be benchmarked as part of the selection process on either a defined pre-cursor dataset or data taken during LSST commissioning. 
+
+\textbf{Inputs and Outputs:}
+Photo-$z$ estimators should require only catalog-level inputs and not require the use of pixels. 
+Inputs may include all measurements in all filter bands for a given object in the DR {\tt Object} catalog. 
+The DM team will ensure that all measured quantities needed as inputs to photo-$z$ estimators are computed and included in the {\tt Object} catalog (See Appendix~\ref{ssec:dp_objvals}). 
+
+The outputs currently foreseen and budgeted for, irrespective of algorithm, are defined in the LSST Data Products Definition Document (DPDD) \cite{LSE-163}, and are presented in Appendix \ref{ssec:docs_dpdd}. 
+Exact schema definitions for the output user-facing data products are stored in the github repository sdm\_schemas (\url{https://github.com/lsst/sdm\_schemas}). 
+This repository represents the source of truth for the schema definitions for user-facing data products.
+A detailed description of schema management in DM is presented in \cite{dmtn-153}. 
+Community input on the currently defined defined inputs and outputs is solicited and this topic will be open for discussion during the LSST Photo-$z$ Virtual Forum series. 
+
+Any input measured quantities or outputs that are needed by proposed photo-$z$ estimators and that are not part of the current DR {\tt Object} catalog baseline should be presented for discussion during the LSST Photo-$z$ Virtual Forum series and detailed in the letters of recommendation for photo-$z$ estimators. 
+
+\textbf{Storage Constraints:}
+Additional storage may be required for ancillary input data such as spectral templates, calibration data or training sets (See Appendix \ref{ssec:dp_calib}). 
+Storage for outputs defined in the the DPDD \cite{LSE-163} has already been budgeted for by DM and is presented in Appendix~\ref{ssec:dp_objvals}. 
+Letters of recommendation for photo-$z$ estimators should list all necessary ancillary data together with an estimate of the storage required.
+
```

Comment:
I think we probably want a dedicated section on training and iterative development; that is the technical component that worries me the most.  Here's some draft text to kick that off:
```suggestion

\textbf{Training and Iterative Development:}
Photo-$z$ estimators often require training of some kind before use on a particular dataset, or even a particular reprocessing of a particular dataset, in order to accurately relate the detailed properties and tendencies of the input photometric measurements to external data.
This is rarely something that can be fully automated.
Our ideal scenario is a ``pre-trained'' estimator that can be run at scale on the full {\tt Object} catalog with no human intervention.
This would typically include both software and auxilliary data that represents (and probably condenses) the results of any training procedure.

To construct such an estimator, we understand that it will probably be necessary to actually train it with photometry that is representative of the target data release, on fields that include at least some of the spectroscopic or external many-band photometry the estimator is trained against.
We plan to perform preliminary processing of small areas of sky prior to beginning full data release processing in earnest, in order to validate the software to be used in that processing.
We expect to make the photometry from this processing available to photo-$z$ collaborators to provide an opportunity for interactive training and tuning before the estimator and its auxiliary data are finalized.
The time window for this work will be short, however, and coverage of important fields may be missing, incomplete, or not representative of the survey as a whole.
While the needs of the photo-$z$ estimator will play a role in what preliminary processing is performed, these will need to be balanced with time constraints, our ability to use that processing for validation, and possibly data rights considerations as well.
Having a photo-$z$ estimator that can operate robustly in the presence of these limitations is thus extremely important (and more important, for our purposes, than the quality of the estimator under more ideal conditions).

Photo-$z$ estimators that cannot work well under these conditions may of course still be extremely valuable for LSST science, but it probably makes more sense to run these after a data release has gone out, and federate them with it later, to provide enough time for more complete training and validation.

If DM is unable to identify an algorithm (and collaborators) for which fast-turnaround, limited-input training is viable, we may simply run an off-the-shelf algorithm with no training at all, to serve as a placeholder until a better post-DR, community-built federated catalog becomes available.
```

This raises a few questions of its own:
 - should we ask people making letters of recommendation to talk about _who_ might do interactive training work?
 - can we actually give non-staff people doing interactive training work access to the preliminary process, in a data-rights sense?
 - are also accepting letters of recommendation or otherwise trying to start collaboration on photo-$z$ estimators that are _not_ suitable for release with a DR, but are instead always going to be run afterwards?

Reply:
Oh, wow, that was much longer than your other subsections - please feel free to trim as you like, no hard feelings :-)

Reply:
Q1: added note that we expect estimators to be supported and that we will work with the community to identify who will do this for the short-listed estimators
Q2: This will require an update to the data rights document, I propose that we approach this as for community contributions to commissioning 
Q3: No. This work is aimed at fulfilling our construction requirements of providing an estimate of photometric redshift in the Object catalog. We will address this is (very important goal) as part of operations.